src/FriendsManagementClient.ts

```diff
@@ -88,10 +88,22 @@ export class FriendsManagementClient implements FriendsManagementAPI {
             .filter(([, status]) => FriendsManagementClient.PENDING_STATUSES.includes(status))
             .map(([room, status]) => {
                 const other = room.guessDMUserId()
+                const event = getLastFriendshipEventInRoom(room)
```

Comment:
I think we need to be sure here that the event is the `request` one, so we probably want to ask for the event of the requester. Most of the time getting the last is enough but not completely sure will always work.